sklearn/calibration.py

```diff
@@ -202,11 +188,47 @@ def fit(self, X, y, sample_weight=None):
             base_estimator = self.base_estimator
 
         if self.cv == "prefit":
+            # Set `n_features_in_` attribute
+            try:
+                check_is_fitted(self)
+            except NotFittedError as nfe:
+                raise AttributeError(
+                    "{} object has no n_features_in_ attribute."
+                    .format(self.__class__.__name__)
+                ) from nfe
+            else:
+                try:
+                    self.n_features_in_ = self.base_estimator.n_features_in_
+                except AttributeError:
+                    pass
+            self.classes_ = base_estimator.classes_
+
             calibrated_classifier = _CalibratedClassifier(
                 base_estimator, method=self.method)
             calibrated_classifier.fit(X, y, sample_weight)
             self.calibrated_classifiers_.append(calibrated_classifier)
         else:
+            X, y = self._validate_data(
+                X, y, accept_sparse=['csc', 'csr', 'coo'],
+                force_all_finite=False, allow_nd=True
+            )
+            le = LabelBinarizer().fit(y)
+            self.classes_ = le.classes_
+
+            # Check that each cross-validation fold can have at least one
+            # example per class
+            if isinstance(self.cv, int):
+                n_folds = self.cv
+            elif hasattr(self.cv, "n_folds"):
+                n_folds = self.cv.n_folds
+            else:
+                n_folds = None
+            if n_folds and np.any([np.sum(y == class_) < n_folds
+                                  for class_ in self.classes_]):
```

Comment:
FYI couldn't use temporary var to store `np.any([....` because it doesn't work when `n_folds` is `None`. (but works well in the `and` statement since it wouldn't be evaluated if `n_folds` is `None`)